v3/src/data-interactive/data-interactive-data-set-types.ts

```diff
@@ -114,3 +114,5 @@ export interface DIUpdateItemResult {
 export interface DIDeleteCollectionResult {
   collections?: number[]
 }
+
+export type DICategoryColorMap = Record<string, string | undefined>
```

Comment:
Can you explain why you need the `undefined` here? It seems like it's unnecessary to me, and it would simplify your code elsewhere.

Reply:
I think the API question here is what is the mechanism for removing a color assignment from the color map? I think you're right that `undefined` is unnecessary, but `null` may be useful for indicating that a color should be removed from the color map (and thus returned to its default color). Alternatively, we could say that `""` is used for that purpose. As I think about it, using `""` seems reasonable from an API perspective and simpler to implement so I'd second @tealefristoe's suggestion of simply removing `undefined` from the definition.